supervise.go

```diff
@@ -554,12 +555,49 @@ func killSupervisedServices() {
 		s.state.Set(Stopping)
 
 		s.setStopped(true)
-		s.Signal(syscall.SIGTERM)
+		s.Signal(signal)
+		states = append(states, s.state)
 	}
+	return states
+}
 
-	for _, s := range services.S {
-		s.waitTillStopped()
+// killSupervisedServices is called before rebooting when upgrading, allowing
+// processes to terminate in an orderly fashion.
+func killSupervisedServices(signalDelay time.Duration) {
```

Comment:
I know I said in the issue that there should be a time after which shutdown should proceed. I had forgotten about this existing SIGTERM code, which blocks until the process exits. 

I think that’s actually the better model (at least by default), as it exposes the hanging shutdown to the user (ideally during development when it’s obvious where the problem comes from) who is then incentivized to fix it :)

Can you adjust the code to keep waiting indefinitely please?

Reply:
Oh, I just now realize that the 15 seconds timeout is already in the existing code. Nevertheless, I think it might be good to make this timeout configurable. On the machine that runs my VM, I would want to configure it way higher than just 15 seconds (as the VM might take a while to shut down, too).